blocks/blockstest/chain.go

```diff
@@ -28,12 +31,53 @@ func NewChainBuilder(genesis *blocks.Block) *ChainBuilder {
 	}
 }
 
+// A ChainOption configures [ChainBuilder.NewBlock].
+type ChainOption = options.Option[chainOptions]
+
+// SetDefaultOptions sets the default options upon which all
+// additional options passed to [ChainBuilder.NewBlock] are appended.
+func (cb *ChainBuilder) SetDefaultOptions(opts ...ChainOption) {
+	cb.opts = opts
+}
+
+type chainOptions struct {
+	ethOpts []EthBlockOption
+	saeOpts []BlockOption
+}
+
+// WithEthBlockOptions wraps the options that [ChainBuilder.NewBlock] propagates
+// to [NewEthBlock].
+func WithEthBlockOptions(opts ...EthBlockOption) ChainOption {
+	return options.Func[chainOptions](func(co *chainOptions) {
+		co.ethOpts = append(co.ethOpts, opts...)
+	})
+}
+
+// WithBlockOptions wraps the options that [ChainBuilder.NewBlock] propagates to
+// [NewBlock].
+func WithBlockOptions(opts ...BlockOption) ChainOption {
+	return options.Func[chainOptions](func(co *chainOptions) {
+		co.saeOpts = append(co.saeOpts, opts...)
+	})
+}
+
+func ethBlockOptions(opts []ChainOption) []EthBlockOption {
+	return options.ApplyTo(&chainOptions{}, opts...).ethOpts
+}
+
+func blockOptions(opts []ChainOption) []BlockOption {
+	return options.ApplyTo(&chainOptions{}, opts...).saeOpts
+}
+
 // NewBlock constructs and returns a new block in the chain.
-func (cb *ChainBuilder) NewBlock(tb testing.TB, txs []*types.Transaction, opts ...EthBlockOption) *blocks.Block {
+func (cb *ChainBuilder) NewBlock(tb testing.TB, txs []*types.Transaction, opts ...ChainOption) *blocks.Block {
 	tb.Helper()
+	opts = slices.Concat(cb.opts, opts)
+
 	last := cb.Last()
-	eth := NewEthBlock(last.EthBlock(), txs, opts...)
-	cb.chain = append(cb.chain, NewBlock(tb, eth, last, nil)) // TODO(arr4n) support last-settled blocks
+	eth := NewEthBlock(last.EthBlock(), txs, ethBlockOptions(opts)...)
+	cb.chain = append(cb.chain, NewBlock(tb, eth, last, nil, blockOptions(opts)...)) // TODO(arr4n) support last-settled blocks
+
 	return cb.Last()
 }
```

Comment:
optional (and pedantic), but do `ethBlockOptions` and `blockOptions` really improve readability?

I personally find the layers of indirection less clear:
```suggestion
// NewBlock constructs and returns a new block in the chain.
func (cb *ChainBuilder) NewBlock(tb testing.TB, txs []*types.Transaction, opts ...ChainOption) *blocks.Block {
tb.Helper()
	opts = slices.Concat(cb.opts, opts)
	options := options.ApplyTo(&chainOptions{}, opts...)

last := cb.Last()
	eth := NewEthBlock(last.EthBlock(), txs, options.ethOpts...)
	cb.chain = append(cb.chain, NewBlock(tb, eth, last, nil, options.saeOpts...)) // TODO(arr4n) support last-settled blocks

return cb.Last()
}
```